Add pylontech integration (#2841)
OttoWinter/esphomedocs: components/pylontech.rst

Any connections via CAN or RS485 (e.g. to an inverter) are untouched and remain functional.

The console port offers a RS232 interface using a RJ10 or RJ45 connector.
The voltage levels are *not* TTL-compatible. A RS232 transceiver must be placed between the Batteries and the ESPHome device.
MAX3232-based transceivers have been tested and work well.

If you have multiple batteries you need to connect to the master battery's console port.

.. list-table:: Pylontech RJ45 Console Port (US2000C, US3000C)
    :header-rows: 1

    * - RJ45 Pin
      - Function
      - Connect to
    * - 3
      - Pylontech TX
      - ESPHome RX via transceiver
    * - 6
      - Pylontech RX
      - ESPHome TX via transceiver
    * - 8
      - GND
      - GND

.. figure:: images/rj45_pinout.jpg
    :align: center
    :width: 70.0%

    RJ45 pinout

.. list-table:: Pylontech RJ10 Console Pinout (US2000B)
    :header-rows: 1

    * - RJ410 Pin
      - Function
      - Connect to
    * - 2
      - Pylontech RX
      - ESPHome TX via transceiver
    * - 3
      - Pylontech TX
      - ESPHome RX via transceiver
    * - 4
      - GND
      - GND
      

> Configuration

Reviewer: ```suggestion
Component/Hub
```

-------------

.. code-block:: yaml

>     # Example configuration entry
>     uart:
>       - id: pylontech_uart
>         tx_pin: GPIO1
>         rx_pin: GPIO3
>         baud_rate: 115200
> 

Reviewer: We don't add external examples here. It is up to the user to visit the uart docs to see how to configure that for their use case.
```suggestion
    # Example configuration entry
```

>     pylontech:
>       - uart_id: pylontech_uart
>         id: pylontech0
>         update_interval: 5s

Reviewer: Examples only needs the most basic configuration required.
```suggestion
    pylontech:
```

Configuration variables:
~~~~~~~~~~~~~~~~~~~~~~~~

- **id** (**Required**, :ref:`config-id`): The id to use for this Pylontech component.
- **uart_id** (*Optional*): The uart Bus ID
- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the sensor. Defaults to ``60s``.

Sensor
------

All values are reported for every Pylontech battery individually.

.. code-block:: yaml

    # Example configuration entry
    sensor:
      - platform: pylontech

>         pylontech_id: pylontech0

Reviewer: When the example is used with only one component this is not needed
```suggestion
```

        battery: 1
        

>         voltage:
>           id: bat1_voltage
>           name: "Battery1 Voltage"
>         current:
>           id: bat1_current
>           name: "Battery1 Current"
>         coulomb:
>           id: bat1_soc
>           name: "Battery1 State of Charge"

Reviewer: ```suggestion
        voltage:
          name: "Battery1 Voltage"
        current:
          name: "Battery1 Current"
        coulomb:
          name: "Battery1 State of Charge"
```

>           
>           
>       - platform: pylontech
>         pylontech_id: pylontech0
>         battery: 2
>         
>         voltage:
>           id: bat2_voltage
>           name: "Battery2 Voltage"
>         current:
>           id: bat2_current
>           name: "Battery2 Current"
>         coulomb:
>           id: bat2_soc
>           name: "Battery2 State of Charge"

Reviewer: ```suggestion
```

Configuration variables:
~~~~~~~~~~~~~~~~~~~~~~~~

> - **pylontech_id** (**Required**): ID of the pylontech instance

Reviewer: ```suggestion
- **pylontech_id** (*Optional*): Manually specify the ID of the pylontech instance if there are multiple.
```

> - **battery** (**Required**): Which battery to monitor. 1 stands for the master battery, 2..6 for slave batteries.

Reviewer: We try not to use master/slave terms
```suggestion
- **battery** (**Required**): Which battery to monitor. 1 stands for the main battery, 2..6 for child batteries.
```

> - **voltage** (*Optional*): Voltage. Should be around 48V

Reviewer: Also needs to apply to the following sensors and similar for text sensors.
```suggestion
- **voltage** (*Optional*): Voltage of the battery. All options from :ref:`Sensor <config-sensor>`.
```

- **current** (*Optional*): Current
- **coulomb** (*Optional*): State of Charge in percent
- **temperature** (*Optional*): Temperature
- **temperature_low** (*Optional*): Historic minimum temperature
- **temperature_high** (*Optional*): Historic maximum temperature
- **voltage_low** (*Optional*): Historic minimum voltage
- **voltage_high** (*Optional*): Historic maximum voltage
- **mos_temperature** (*Optional*): Temperature of the mosfets.

> All sensors are normal sensors, so each one supports options from :ref:`Sensor <config-sensor>`.

Reviewer: Inline is better as it shows up with code completion dialogs
```suggestion
```
